meadow-client/src/MainFrame.purs

```diff
@@ -251,13 +252,13 @@ updateOracles cbn (State state) inputs omap =
 updateActions :: MarloweState -> {state :: State, contract :: Contract, outcome :: TransactionOutcomes, validity :: TransactionValidity} -> MarloweState
 updateActions oldState {state, contract, outcome, validity} =
   set (_input <<< _inputs) (scoutPrimitives oldState.blockNum state contract)
-  (over (_input <<< _choiceData) (updateChoices state neededInputs)
-  (over (_input <<< _oracleData) (updateOracles oldState.blockNum state neededInputs)
+  (over (_input <<< _choiceData) (updateChoices state neededInputs')
+  (over (_input <<< _oracleData) (updateOracles oldState.blockNum state neededInputs')
   (set (_transaction <<< _outcomes) outcome
   (set (_transaction <<< _validity) validity
    oldState))))
   where
-    neededInputs = collectNeededInputsFromContract contract
+    neededInputs' = neededInputs contract
```

Comment:
This is confusing - one's a value and one's a function? I suggest giving `neededInputs'` a better name.

Reply:
indeed it is but I just couldn't think of a decent name. will try again

Reply:
changed function to `collectNeededInputs`